### ISSUE TITLE
19 Dynamo test are failing with "Sdpa_Shape.__new__() missing 3 required positional arguments: 'num_heads', 'seq_len', and 'head_dim'".

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#119785]
19 Dynamo test are failing with "Sdpa_Shape.__new__() missing 3 required positional arguments: 'num_heads', 'seq_len', and 'head_dim'".

## Repro

`PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_causal_variant_1_shape3_cpu`

You will need to remove the skip or expectedFailure before running the repro command.
This may be just removing a line in
[dynamo_test_failures.py](https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/dynamo_test_failures.py)

## Failing tests

Here's a comprehensive list of tests that fail (as of this issue) with the above message:
<details>
<summary>Click me</summary>
```
test_transformers.py::TestAttnBiasCPU.test_causal_variants_causal_variant_1_shape3_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_causal_variant_2_shape0_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_compile_causal_variant_1_shape0_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_compile_causal_variant_1_shape2_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_compile_causal_variant_1_shape3_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_compile_causal_variant_2_shape1_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_compile_causal_variant_2_shape3_cpu
test_transformers.py::TestAttnBiasCPU.test_is_causal_equals_upper_left_shape0_cpu
test_transformers.py::TestAttnBiasCPU.test_is_causal_equals_upper_left_shape1_cpu
test_transformers.py::TestAttnBiasCPU.test_is_causal_equals_upper_left_shape2_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_causal_variant_1_shape0_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_causal_variant_1_shape1_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_causal_variant_1_shape2_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_causal_variant_2_shape1_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_causal_variant_2_shape3_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_compile_causal_variant_1_shape1_cpu
test_transformers.py::TestAttnBiasCPU.test_causal_variants_compile_causal_variant_2_shape0_cpu
test_transformers.py::TestAttnBiasCPU.test_is_causal_and_mask_fails_cpu
test_transformers.py::TestAttnBiasCPU.test_is_causal_equals_upper_left_shape3_cpu
```

cc @ezyang @msaroufim @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

**Comment by y-sq** [comment]
The failed tests either don't exit or can pass on my local environment. Here is my results of re-running them:
```
(pytorch) [shuqiyang@devgpu001.lla3 ~/local/pytorch/test (main)]$ bash ./test_list.sh 
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_causal_variant_1_shape3_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 15.52s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_causal_variant_2_shape0_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 17.68s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_compile_causal_variant_1_shape0_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 16.98s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_compile_causal_variant_1_shape2_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 16.23s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_compile_causal_variant_1_shape3_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 15.66s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_compile_causal_variant_2_shape1_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 15.48s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_compile_causal_variant_2_shape3_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 15.08s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_is_causal_equals_upper_left_shape0_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77249 deselected / 1 selected                                                                                                               
Running 1 items in this shard

test_transformers.py::TestAttnBiasCPU::test_is_causal_equals_upper_left_shape0_cpu PASSED [0.8126s]                                                           [100%]

=============================================================== 1 passed, 77249 deselected in 17.28s ================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_is_causal_equals_upper_left_shape1_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77249 deselected / 1 selected                                                                                                               
Running 1 items in this shard

test_transformers.py::TestAttnBiasCPU::test_is_causal_equals_upper_left_shape1_cpu PASSED [0.8209s]                                                           [100%]

=============================================================== 1 passed, 77249 deselected in 17.88s ================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_is_causal_equals_upper_left_shape2_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77249 deselected / 1 selected                                                                                                               
Running 1 items in this shard

test_transformers.py::TestAttnBiasCPU::test_is_causal_equals_upper_left_shape2_cpu PASSED [1.4720s]                                                           [100%]

=============================================================== 1 passed, 77249 deselected in 15.89s ================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_causal_variant_1_shape0_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 15.95s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_causal_variant_1_shape1_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 17.29s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_causal_variant_1_shape2_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 16.17s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_causal_variant_2_shape1_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 15.54s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_causal_variant_2_shape3_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 16.97s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_compile_causal_variant_1_shape1_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 14.55s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_causal_variants_compile_causal_variant_2_shape0_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77250 deselected / 0 selected                                                                                                               
Running 0 items in this shard

==================================================================== 77250 deselected in 15.16s =====================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_is_causal_and_mask_fails_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77249 deselected / 1 selected                                                                                                               
Running 1 items in this shard

test_transformers.py::TestAttnBiasCPU::test_is_causal_and_mask_fails_cpu PASSED [3.8620s]                                                                     [100%]

=============================================================== 1 passed, 77249 deselected in 22.47s ================================================================
====================================================================
PYTORCH_TEST_WITH_DYNAMO=1 pytest test_transformers.py -v -k test_is_causal_equals_upper_left_shape3_cpu
======================================================================== test session starts ========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.0.0 -- /home/shuqiyang/.conda/envs/pytorch/bin/python
cachedir: .pytest_cache
rootdir: /data/users/shuqiyang/pytorch
configfile: pytest.ini
collected 77250 items / 77249 deselected / 1 selected                                                                                                               
Running 1 items in this shard

test_transformers.py::TestAttnBiasCPU::test_is_causal_equals_upper_left_shape3_cpu PASSED [0.6269s]                                                           [100%]

=============================================================== 1 passed, 77249 deselected in 18.02s ================================================================
====================================================================
```

**Comment by zou3519** [comment]
We might have just renamed the tests. Running `PYTORCH_TEST_WITH_DYNAMO=1 pytest test/test_transformers.py -v -k "test_causal_variants"`, I see the following test suppressions.

It seems like we can remove many of the test suppressions but there are still some errors, so there's still some work to be done here.
```
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_causal_variant_CausalVariant_LOWER_RIGHT_shape0_cpu SKIPPED [0.9753s] (This test passed, ...) [  3%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_causal_variant_CausalVariant_LOWER_RIGHT_shape1_cpu SKIPPED [0.8209s] (This test passed, ...) [  6%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_causal_variant_CausalVariant_LOWER_RIGHT_shape2_cpu SKIPPED [0.0370s] (Lower right causal...) [  9%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_causal_variant_CausalVariant_LOWER_RIGHT_shape3_cpu SKIPPED [0.5270s] (This test passed, ...) [ 12%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_causal_variant_CausalVariant_UPPER_LEFT_shape0_cpu SKIPPED [0.6085s] (This test passed, m...) [ 15%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_causal_variant_CausalVariant_UPPER_LEFT_shape1_cpu SKIPPED [0.7599s] (This test passed, m...) [ 18%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_causal_variant_CausalVariant_UPPER_LEFT_shape2_cpu SKIPPED [0.8170s] (This test passed, m...) [ 21%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_causal_variant_CausalVariant_UPPER_LEFT_shape3_cpu SKIPPED [0.4809s] (This test passed, m...) [ 25%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_compile_causal_variant_CausalVariant_LOWER_RIGHT_shape0_cpu SKIPPED [1.6867s] (Scalars ar...) [ 28%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_compile_causal_variant_CausalVariant_LOWER_RIGHT_shape1_cpu SKIPPED [0.8996s] (Scalars ar...) [ 31%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_compile_causal_variant_CausalVariant_LOWER_RIGHT_shape2_cpu SKIPPED [0.0505s] (Lower righ...) [ 34%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_compile_causal_variant_CausalVariant_LOWER_RIGHT_shape3_cpu SKIPPED [0.6180s] (Scalars ar...) [ 37%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_compile_causal_variant_CausalVariant_UPPER_LEFT_shape0_cpu SKIPPED [0.6658s] (Scalars are...) [ 40%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_compile_causal_variant_CausalVariant_UPPER_LEFT_shape1_cpu SKIPPED [0.8604s] (Scalars are...) [ 43%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_compile_causal_variant_CausalVariant_UPPER_LEFT_shape2_cpu SKIPPED [0.9052s] (Scalars are...) [ 46%]
test/test_transformers.py::TestAttnBiasCPU::test_causal_variants_compile_causal_variant_CausalVariant_UPPER_LEFT_shape3_cpu SKIPPED [0.5754s] (Scalars are...) [ 50%]
```

**Comment by janeyx99** [comment]
Seems to be fixed by #121048 thanks @y-sq!
